src/mlpack/methods/ann/layer/softmax_impl.hpp

```diff
@@ -44,12 +43,7 @@ void Softmax<InputDataType, OutputDataType>::Backward(
     const arma::Mat<eT>& gy,
     arma::Mat<eT>& g)
 {
-  g.set_size(input.n_rows, input.n_cols);
-  for (size_t i = 0; i < input.n_cols; ++i)
-  {
-    g.col(i) = input.col(i) % (gy.col(i)
-        - arma::as_scalar(gy.col(i).t() * input.col(i)));
-  }
+  g = input % (gy - arma::repmat(arma::sum(gy % input), input.n_rows, 1));
 }
```

Comment:
@zoq: I think this is fine now. What do you think?

Reply:
Looks good.